Xamarin.Forms.Core/Setter.cs

```diff
@@ -47,59 +48,73 @@ object IValueProvider.ProvideValue(IServiceProvider serviceProvider)
 			return this;
 		}
 
-		internal void Apply(BindableObject target, bool fromStyle = false)
+		internal void Apply(BindableObject scope, bool fromStyle = false)
 		{
-			if (target == null)
-				throw new ArgumentNullException(nameof(target));
+			var targetObject = FindTargetObject(scope, TargetName);
```

Comment:
I think the FindTargetObject doesn't bring enough value by itself
```csharp
if (target == null) throw...;
if (!string.IsNullOrEmpty(TargetName) && target is Element element)
    target == element.FindByName(TargetName) as BindableObject ?? throw ...;
```

Reply:
I think I implemented the way you intended, if I missed anything please let me know :)